controllers/cluster_create.go

```diff
@@ -968,39 +968,28 @@ func (r *ClusterReconciler) createPrimaryInstance(
 		return ctrl.Result{}, nil
 	}
 
-	// Generate a new node serial
-	nodeSerial, err := r.generateNodeSerial(ctx, cluster)
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("cannot generate node serial: %w", err)
-	}
-
 	var backup *apiv1.Backup
 	if cluster.Spec.Bootstrap != nil &&
-		cluster.Spec.Bootstrap.Recovery != nil &&
-		cluster.Spec.Bootstrap.Recovery.Backup != nil {
+		cluster.Spec.Bootstrap.Recovery != nil {
+		var err error
 		backup, err = r.getOriginBackup(ctx, cluster)
 		if err != nil {
 			return ctrl.Result{}, err
 		}
```

Comment:
this is already there before your PR, but the flow is very confusing.
We look for a backup object in the bootstrap.recovery, and it looks like we depend on it.
But further down we may still proceed if we had boostrap.recovery.volumesnapshots.
I'm going to try to make this more explicit.